### DISCUSSION TITLE
Pointer to incomplete class "PhysicsBodyData"

### DISCUSSION THREAD

**Discussion by louishamelers** [CapsCollective/raylib-cpp-starter#11] [Q&A]
I'm trying to access the PhysicsBody properties but I keep getting the "Pointer to incomplete clas"  error:

![image](https://user-images.githubusercontent.com/25366619/130589517-d3b8fa25-76b8-48d5-914e-a1db8ffdf972.png)
![image](https://user-images.githubusercontent.com/25366619/130588944-7d9f6c34-e3d9-4414-81c4-f380ea48a47b.png)

My character.hpp has `#include <physics.hpp>`  and defines `PhysicsBody physicsBody`. The weird thing is that I have a seperate player class defined in a .hpp file which looks like this:

![image](https://user-images.githubusercontent.com/25366619/130590634-870378f0-e1f3-4c99-9563-26dbceaa3dca.png)

Which seems to work without a problem...

I have no clue where this problem could originate from, any and all help is very much appreciated. You can see the full repository [here](https://github.com/louishamelers/game/tree/game10) for more details.

Thanks in advance. Cheers!

**Comment by Raelr** [comment]
Hey @louishamelers , 

Thanks for opening up this discussion! I gave your repo a quick look and it looks like you're using two different types in these two files. In the case of `Player.hpp`, you're using a [`PhysicsBody` type](https://github.com/louishamelers/game/blob/game10/src/headers/Player.hpp#L24). In `Character.hpp`, you're using a [`PhysicsBodyData*` type](https://github.com/louishamelers/game/blob/game10/src/headers/characterStateMachine/character.hpp#L18). 

I gave the `raylib` source code a look to find what these are and it looks like `PhysicsBodyData` is [defined twice in the code](https://github.com/raysan5/raylib/blob/master/src/extras/physac.h#L115). It looks like `PhysicsBodyData` is defined as a pointer to a `PhysicsBody`, but is then also re-defined into it's own struct. I'm unsure why this has been done, but I have a feeling this is part of the reason why this is breaking. 

I suspect the most immediate way to resolve this would be to simply use a `PhysicsBody` in your `Character` class instead of a `PhysicsBodyData` struct. 

If that doesn't work then I'd suggest opening up an issue directly with [`raylib`](https://github.com/raysan5/raylib), since this seems to be an issue with their definitions, rather than a build issue. 

I hope this helps narrow down the issue!

**Reply by louishamelers** [reply]
I gave your solution a try, but to no avail. It still gives me the exact same error, which is weird since I don't reference the PhysicsBodydata type anymore... Anyway, i'm submitting an issue on the raylib page. thanks for your help!
